vmdb/app/models/ems_refresh/parsers/openstack_infra.rb

```diff
@@ -153,6 +153,9 @@ def get_extra_host_attributes!(host, hostname)
         host_for_ssh = HostOpenstackInfra.new(:hostname => hostname)
         host_for_ssh.ext_management_system = @ems
 
+        # Skip ssh refresh when there are no credentials
+        return unless host_for_ssh.auth_user_keypair(:ssh_keypair).try(:last)
```

Comment:
Can we throw a query method on the Host, `refreshable_via_ssh?` that does this? similar to how connect_ssh abstracts these details.  I think this would read better and you wouldn't need the comment:

`return unless host_for_ssh.refreshable_via_ssh?`